### PR TITLE
Upgrade Phoenix to 5.1.3

### PR DESCRIPTION
**Pull request by wendigo**
## Release notes

( ) This is not user-visible or docs only and no release notes are required.
( ) Release notes are required, please propose a release note for me.
( ) Release notes are required, with the following suggested text:

```markdown
# Section
* Fix some things. ({issue}`issuenumber`)
```

### CODE REVIEW

**Comment by wendigo** [comment]
This is ready to be review. Phoenix 5.1.3 fixes NULL binary handling, which I've also fixed in our tests.

File: plugin/trino-phoenix5/pom.xml [hunk 1]
```diff
@@ -133,6 +133,14 @@
             </exclusions>
         </dependency>
 
+        <!-- Phoenix depends on log4j api and since it's banned, let's use hardened API-compatible fork -->
+        <dependency>
+            <groupId>ch.qos.reload4j</groupId>
+            <artifactId>reload4j</artifactId>
+            <version>1.2.25</version>
+            <scope>runtime</scope>
+        </dependency>
```
Comment (hashhar) [inline plugin/trino-phoenix5/pom.xml:142]: seems very brittle

File: plugin/trino-phoenix5/src/test/java/io/trino/plugin/phoenix5/TestPhoenixConnectorTest.java [hunk 2]
```diff
@@ -615,6 +616,7 @@ public void testCreateTableWithProperties()
                         "   d varchar(10)\n" +
                         ")\n" +
                         "WITH (\n" +
+                        "   bloomfilter = 'ROW',\n" +
```
Comment (hashhar) [inline plugin/trino-phoenix5/src/test/java/io/trino/plugin/phoenix5/TestPhoenixConnectorTest.java:619]: where does this come from?

Comment (wendigo) [reply plugin/trino-phoenix5/src/test/java/io/trino/plugin/phoenix5/TestPhoenixConnectorTest.java:619]: it's now being returned as a default table property (bloomfilter with all its values was already supported)

File: plugin/trino-phoenix5/pom.xml (hunk 1 above)
Comment (wendigo) [reply plugin/trino-phoenix5/pom.xml:142]: What's the alternative? Ideas?

Comment (hashhar) [reply plugin/trino-phoenix5/pom.xml:142]: Actually looks like Phoenix should already be pulling in reload4j and the slf4j shim for it. See https://github.com/apache/phoenix/commit/cd35c5239bd7e3f263d3f2a99ce1b996205bde4b but I don't know what version that commit is a part of.

Comment (wendigo) [reply plugin/trino-phoenix5/pom.xml:142]: I'm afraid it's not the case for 5.1.3 :(

**Comment by hashhar** [comment]
I just added a TODO to remind when it's time to remove the dependency.

Will merge once CI finishes.

**Comment by wendigo** [comment]
Thx @hashhar

**Pull request merged**